File: phoneList.cpp
#include<string>
#include<iostream> 
#include<vector>
#include <algorithm>
using namespace std;
int main()
{
	int cases, conSize;
	cin >> cases;
	vector<string> contacts;
	string temp;
	bool valid;
	for (int i = 0; i < cases; i++) {
		valid = true;
		cin >> conSize;
		contacts.clear();
		for (int x = 0; x < conSize; x++) {
			cin >> temp;
			contacts.push_back(temp);
		}
		sort(contacts.begin(), contacts.end());
		for (int x = 0; x < conSize-1; x++) {
				if (contacts[x] == contacts[x + 1].substr(0, contacts[x].size())) {
					valid = false;
					break;
				}
		}
		if (valid) {
			cout << "YES" << endl;
		}
		else {
			cout << "NO" << endl;
		}
	}

	return 0;
}

File: phoneList.py
cases = int(input())
contacts = []
for i in range(cases):
    valid = True
    contacts.clear()
    conSize = int(input())
    for x in range(conSize):
        contacts.append(input())
    for x in range(conSize-1):
        for y in range(x+1,conSize):
            if(contacts[y].startswith(contacts[x])):
                x = conSize
                y = conSize
                valid = False
    if(valid):
        print("YES")
    else:
        print("NO")
